src/strongbox/addon.clj

```diff
@@ -116,18 +117,27 @@
         addon-list (apply conj (mapv expand addon-groups) unknown-grouping)]
     addon-list))
 
-(defn-spec ungroup-addon :addon/installed-list
-  "an addon may actually be many addons bundled together with a primary one chosen to represent them.
-  sometimes we want to treat this addon as a list of addons"
-  [addon :addon/installed]
-  (if (empty? (:group-addons addon))
-    [addon]
-    (get addon :group-addons [])))
+(defn merge-source-map-lists
```

Comment:
there isn't anything source-map specific about this fn. perhaps stick it in utils?

tests

defn-spec